src/coreclr/vm/methodtable.h

```diff
@@ -2998,10 +3029,11 @@ public :
             MethodTable* pMT;
         };
 
-        static void* operator new(size_t size, TargetMethodTable targetMT)
+        static void* operator new(size_t size, TargetMethodTable targetMT, MethodDataComputeOptions computeOptions)
         {
-            _ASSERTE(size <= GetObjectSize(targetMT.pMT));
-            return ::operator new(GetObjectSize(targetMT.pMT));
+            _ASSERTE(computeOptions != MethodDataComputeOptions::CacheOnly);
+            _ASSERTE(size <= GetObjectSize(targetMT.pMT, computeOptions));
```

Comment:
Why is it okay for the size to be less than?

Reply:
This is a C++ wonkiness. `size` is the `sizeof(MethodDataObject)` passed to operator new implicitly by the new expression. We are asserting here that `GetObjectSize` never suggests we should allocate less that `sizeof(MethodDataObject)` amounts of data.